testim-created/unzip-files-validation.js

```diff
@@ -0,0 +1,143 @@
+example ['Test','This is a test']
+
+NOTE: 'Test' is the content of the first file and 'This is a test' is the content of the second file
+
+ encoding
+ <NOTE for text files use encoding of 'UTF-8' for .jpg file is Base64> Check type of file for encoding needed is 
+ it will vary 
+
+ Optional parameters:
+
+This is an CLI action step in testim.io so make sure the agent is running 
+ in another terminal window if running locally. Command to run is as follows:
+
+ npm i -g @testim/testim-cli && testim connect
+ 
+ NOTE code here is AS-IS and you are own your own in terms of support.
+
+ save your changes and run!
+*/
+
+// Define variables
+var fs = require('fs');
+let filename = `${filedir}${fileName}`;
+let zipEntries = [];
+let newArray = [];
+let results = [];
+
+// check if required parameters are empty or undefined and throw an error 
+
+if (typeof filedir === '' || typeof filedir === 'undefined') {
+    throw new Error('Enter a file dirctory');
+}
+
+if (typeof encoding === '' || typeof encoding === 'undefined') {
+    throw new Error('Enter valid encoding string');
+}
+// extract zip file and load into an array of records (files)
+console.log('File name is ' +filename);
+let zip = new AdmZip(filename);
+zipEntries = zip.getEntries();   
+
+// zip entries is an object so to compare files there is only need for the file name 
+// get file name and load into new array
+zipEntries.forEach(file => newArray.push(file.name)); 
+
+// perform valiation loop over array of extracted files
+for (let i=0; i<= newArray.length; i++) {
+
+     // if the extracted file name matches with our expected file name then set the result to true and add  to the results array
+     // otherwise set the result to false
+     
+    if (newArray[i] === fileArray[i]) {
+        console.log(`File from extracted zip ${newArray[i]} is the same as file to validate: ${fileArray[i]}`);
+        results.push(true);
+    } else {
+         console.log(`File from extracted zip ${newArray[i]} is not the same as file to validate: ${fileArray[i]}`);
+         results.push(false);
+    }
+}
+
+// print out file content which could be used for validation of what is in a file as well expectedFileContent
+// if there is a need to loop all files uncomment 
+
+/*
+for (let i=0; i<= zipEntries.length; i++ ) {
+
+    let extractedFileContent = zipEntries[i].getData().toString(encoding);
+    console.log('Extracted file data: ' +extractedFileContent);
+
+    if (extractedFileContent === expectedFileContent[i]) {
+         results.push(true);
+    } else {
+         console.log(`Extracted data: ${extractedFileContent} did not match expected data: ${expectedFileContent[i]}`);
+         results.push(false);
+    }
+}
+*/
+
+zipEntries.forEach(file => console.log(`File Name ${file.name} File content ${file.getData().toString(encoding)}`));
+
+let myFirstFileContent = zipEntries[0].getData().toString(encoding);
+//console.log('First file content = ' +myFirstFileContent);
+
```

Comment:
If the download is an empty zip file then this will throw an exception as zipEntries[0] will be null.  If we are ok with that then ignore this comment.  Line 109 will error if there is 0 or one file in the zip archive.